pkg/stanza/operator/helper/multiline.go

```diff
@@ -96,13 +96,17 @@ func (c MultilineConfig) getSplitFunc(enc encoding.Encoding, flushAtEOF bool, fo
 
 // NewLineStartSplitFunc creates a bufio.SplitFunc that splits an incoming stream into
 // tokens that start with a match to the regex pattern provided
-func NewLineStartSplitFunc(re *regexp.Regexp, flushAtEOF bool) bufio.SplitFunc {
+func NewLineStartSplitFunc(re *regexp.Regexp, flushAtEOF bool, preserveWhitespace bool) bufio.SplitFunc {
 	return func(data []byte, atEOF bool) (advance int, token []byte, err error) {
 		firstLoc := re.FindIndex(data)
 		if firstLoc == nil {
 			// Flush if no more data is expected
 			if len(data) != 0 && atEOF && flushAtEOF {
-				token = trimWhitespaces(data)
+				if preserveWhitespace {
+					token = data
+				} else {
+					token = trimWhitespaces(data)
+				}
```

Comment:
Since we reuse this pattern a lot, maybe we should set the appropriate func:
```go
type trimFunc func([]byte) []byte

func noTrim(token []byte) []byte {
    return token
}

func NewWhateverSplitFunc(..., preserveWhitespace bool) bufio.SplitFunc {
    trim := trimWhitespaces
    if preserveWhiteSpace {
        trim = noTrim
    }

    ...

    token = trim(token)

    ...
}

```

It's not clear to me that this is better, but what do you think?

Reply:
Thanks for the suggestion, I think it's better than if else block, I've update it like this way.